### PR TITLE
Handle new Company field in e-mail accounts

### PR DESCRIPTION
**Pull request by FredericEspiau**
## Handle new Company field in e-mail accounts
MBP-202

### CODE REVIEW

File: src/exchange/account/home/account-home.component.js [hunk 1]
```diff
@@ -148,6 +165,19 @@
           for (let i = formattedAccounts.length; i < this.accounts.length; i += 1) {
             this.accounts.splice(i, 1);
           }
+
+          if (this.preferencesAlreadyExists) {
+            return this.$q.when();
```
Comment (frenautvh) [inline src/exchange/account/home/account-home.component.js]: you can just ```return;```

File: src/exchange/account/home/account-home.component.js [hunk 2]
```diff
@@ -148,6 +165,19 @@
...
+          }
+
+          const newCompanyColumnParameter = this.computeDefaultCompanyColumnParameter();
+          const changesHaveBeenDone = this.computeDatagridColumnParameters(
+            newCompanyColumnParameter,
+          );
+
+          return changesHaveBeenDone
```
Comment (frenautvh) [inline src/exchange/account/home/account-home.component.js]:
```
if (changesHaveBeenDone) {
  return this.savingDatagridColumnParameters();
}
```

File: src/exchange/account/home/account-home.component.js [hunk 3]
```diff
@@ -174,24 +204,42 @@
...
+          );
+          this.datagridData = {
             data: this.accounts,
             meta: {
               totalCount: accounts.count,
             },
           };
+
+          if (this.preferencesAlreadyExists) {
+            return this.$q.when();
```
Comment (frenautvh) [inline src/exchange/account/home/account-home.component.js]: same

File: src/exchange/account/home/account-home.component.js [hunk 4]
```diff
@@ -174,24 +204,42 @@
...
+          }
+
+          const newCompanyColumnParameter = this.computeDefaultCompanyColumnParameter();
+          const changesHaveBeenDone = this.computeDatagridColumnParameters(
+            newCompanyColumnParameter,
+          );
+
+          return changesHaveBeenDone
```
Comment (frenautvh) [inline src/exchange/account/home/account-home.component.js]: same

File: src/exchange/account/update/account-update.controller.js [hunk 5]
```diff
@@ -375,10 +378,9 @@ angular.module('Module.exchange.controllers').controller(
             this.services.messaging.setMessage(this.getActionMessage(data.messages), data);
           })
           .catch((failure) => {
-            this.services.messaging.writeError(
-              this.services.$translate.instant('exchange_common_error'),
-              failure,
-            );
+            this.services.messaging.writeError(`
+              ${this.services.$translate.instant('exchange_common_error')} ${_.get(failure, 'message', failure)}
```
Comment (frenautvh) [inline src/exchange/account/update/account-update.controller.js]: lodash might not be available here

File: src/exchange/account/home/account-home.component.js [hunk 6]
```diff
@@ -276,17 +328,46 @@
...
+              default:
+                return true;
+            }
+          });
+      }
+
+      let formattedAccounts = _.get(accounts, 'list.results', [])
+        .map(account => ({
+          ...account,
+          ...{
```
Comment (marie-j) [inline src/exchange/account/home/account-home.component.js]:
```suggestion

```
You could just write directly the propeties and their values ?

Comment (FredericEspiau) [reply src/exchange/account/home/account-home.component.js]: The whole 50 ?

File: src/exchange/account/home/account-home.component.js (hunk 1 above)
Comment (FredericEspiau) [reply src/exchange/account/home/account-home.component.js]: No I can't but returning `this.$q.when()` in a `Promise` is indeed non-sensical 😄

File: src/exchange/account/home/account-home.component.js (hunk 2 above)
Comment (FredericEspiau) [reply src/exchange/account/home/account-home.component.js]: Nop, ESLINT doesn't like that

**Comment by antleblanc** [comment]
@FredericEspiau Could you please resolve existing conflict? Thanks:)

**Comment by FredericEspiau** [comment]
@antleblanc done

**Comment by FredericEspiau** [comment]
@antleblanc done again 🎉

**Pull request merged**